src/npctalk.cpp

```diff
@@ -4047,9 +4066,15 @@ talk_effect_fun_t::func f_bulk_trade_accept( const JsonObject &jo, std::string_v
         if( tmp.count_by_charges() ) {
             seller->use_charges( d.cur_item, seller_has );
         } else {
-            seller->use_amount( d.cur_item, seller_has );
+            seller->use_amount( d.cur_item, seller_has_loose );
+            seller->remove_items_with( is_canned_item );
```

Comment:
This looks like it removes all canned items from the seller? Do we want to be more specific than that?

Reply:
is_canned_item variable name could be improved, it's a lambda that looks for items that contains a single, sealed pocket, that contains only 1 of the item type we are looking for.

Reply:
ohh, doh, missed the type check at the end of the lambda. cool cool.